release_creation/github_client/releases.py

```diff
@@ -0,0 +1,142 @@
+import copy
+import os
+import requests
+from typing import Dict, List, Optional, Set, Tuple
+from semantic_version import Version
+from github import Github
+from github.GithubException import GithubException
+from github.GitRelease import GitRelease
+from github.GitReleaseAsset import GitReleaseAsset
+
+_GH_SNAPSHOT_REPO = "dbt-labs/dbt-core-snapshots"
+_GH_ACCESS_TOKEN = os.environ["GH_ACCESS_TOKEN"]
+_SNAP_REQ_NAME = "snapshot_requirements"
+
+
+def get_github_client() -> Github:
+    return Github(_GH_ACCESS_TOKEN)
+
+
+def get_latest_snapshot_release(input_version: str) -> Tuple[ Version, Optional[GitRelease]]:
+    """Retrieve the latest release matching the major.minor and release stage
+       semantic version if it exists. Ignores the patch version. 
+
+    Args:
+        input_version (str): semantic version (1.0.0.0rc, 2.3.5) to match against 
+
+    Returns:
+        Tuple[ Version, Optional[GitRelease]]: _description_
+    """
+    gh = get_github_client()
+    target_version = Version.coerce(input_version)
+    latest = copy.copy(target_version)
```

Comment:
naming nit: latest_version would be descriptive / readable here

Reply:
renamed